src/hotspot/os/posix/perfMemory_posix.cpp

```diff
@@ -346,7 +349,11 @@ static DIR *open_directory_secure(const char* dirname) {
   }
 
   // Check to make sure fd and dirp are referencing the same file system object.
+#if defined(_AIX)
+  if (!is_same_fsobject(fd, dirp->dd_fd)) {
+#else
   if (!is_same_fsobject(fd, dirfd(dirp))) {
+#endif
```

Comment:
I think this would be cleaner with AIX_ONLY/NOT_AIX macros.

Reply:
Changed to use AIX_ONLY/NOT_AIX macros.